### PR TITLE
docs: Add commit message guidelines

### PR DESCRIPTION
**Pull request by prmathur-microsoft**
# Checklist
- [X] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [X] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

**Comment by timtay-microsoft** [comment]
I don't love the idea of splitting commits by package. Generally, when we complete a feature that spans across 1 or more packages, it would never make sense to revert the changes to just one of those packages. Doing so would just break the other package.

What is the argument for splitting commits by package?

**Comment by mamokarz** [comment]
Agree with timtay comment.

**Comment by timtay-microsoft** [comment]
From what I can tell, the reason commits are being split by package in these guidelines is so that we have an easy way of checking which packages were updated by each commit. Instead of having one Scope per commit, we should just allow multiple Scopes per commit. That way, we are still defining which packages are affected in a clear way, but we won't have to separate commits by package.

File: .github/CODING_GUIDELINES.md [hunk 1]
```diff
@@ -252,3 +253,109 @@ public class ThisIsASampleClassTest()
...
+
+**Footer**
+
+The footer should contain any information about Breaking Changes and is also the place to reference GitHub issues that this commit Closes.
+
+Breaking Changes should start with the word BREAKING CHANGE: with a space or two newlines. The rest of the commit message is then used for this.
+
+***Example commit messages***
+
+Good commit messages looks like below :
```
Comment (tameraw) [inline .github/CODING_GUIDELINES.md]: nitpic: "look" (and below)

File: .github/CODING_GUIDELINES.md [hunk 2]
```diff
@@ -252,3 +253,109 @@ public class ThisIsASampleClassTest()
...
+The subject contains succinct description of the change:
+
+* use the imperative, present tense: "change" not "changed" nor "changes"
+* no dot (.) at the end
+
+**Body**
+
+Just as in the subject, use the imperative, present tense: 
+
+"change" not "changed" nor "changes". The body should include the motivation for the change and contrast this with previous behavior.
```
Comment (tameraw) [inline .github/CODING_GUIDELINES.md]: Do we need to state that? Can't we point to a link with that common practice?

File: .github/CODING_GUIDELINES.md [hunk 3]
```diff
@@ -252,3 +253,109 @@ public class ThisIsASampleClassTest()
...
+
+    Github issue (fix#123)
+* feat(iot-service): Add continuation token to Query
+* docs: Update readme to reflect provisiong client
+
+Bad commit messages looks like below :
+
+* fix(iot-dev): small fix
+
+    I was trying to reconnect and network dropped. Fixing such random failures
```
Comment (tameraw) [inline .github/CODING_GUIDELINES.md]: >Fixing [](start = 51, length = 6)

Shouldn't be "fix" if I follow the rule above for imperative?

File: .github/CODING_GUIDELINES.md (hunk 1 above)
Comment (prmathur-microsoft) [reply .github/CODING_GUIDELINES.md]: done

File: .github/CODING_GUIDELINES.md (hunk 3 above)
Comment (prmathur-microsoft) [reply .github/CODING_GUIDELINES.md]: This is to show how a bad commit message looks like. Hence "Fixing" is used on purpose.

File: .github/CODING_GUIDELINES.md (hunk 2 above)
Comment (prmathur-microsoft) [reply .github/CODING_GUIDELINES.md]: Will do that

**Comment by prmathur-microsoft** [comment]
As per the conversation changed the following : 
1.Optionally split commits per packages. Just do the needful.
2. just call scope "samples"
3. addressed further comments

**Pull request merged**